test/unit/prof_mdump.c

```diff
@@ -169,7 +173,9 @@ TEST_BEGIN(test_mdump_maps_error) {
 
 	prof_dump_open_file_t *open_file_orig = prof_dump_open_file;
 	prof_dump_write_file_t *write_file_orig = prof_dump_write_file;
+#ifndef __APPLE__
 	prof_dump_open_maps_t *open_maps_orig = prof_dump_open_maps;
+#endif
```

Comment:
Maybe keep the definition of prof_dump_open_maps_t for mac (but set prof_dump_open_maps to NULL or simply calling `NOT_REACHED()`), and then in this test we don't have to #ifdef many times?

The only feedback I have is, we should cut down the number of ifdefs, by grouping them together or making things more "compatible". But honestly I couldn't find a clean way. @nullptr0-0 if you also don't see a good way, we can leave it as is.